crates/re_build_info/src/crate_version.rs

```diff
@@ -1,94 +1,191 @@
-const IS_PRERELEASE_BIT: u8 = 1 << 6;
+    pub const RC: u8 = 0b01000000;
+    pub const ALPHA: u8 = 0b10000000;
+    pub const DEV_ALPHA: u8 = 0b11000000;
+}
 
 /// The version of a Rerun crate.
 ///
-/// Sub-set of semver supporting `major.minor.patch` plus an optional `-alpha.X`.
-///
-/// When parsing, any `+metadata` suffix is ignored.
+/// Sub-set of semver supporting `major.minor.patch-{alpha,rc}.N+dev`.
 ///
-/// Examples: `1.2.3`, `1.2.3-alpha.4`.
+/// The string value of build metadata is not preserved.
 ///
-/// We use `-alpha.X` when we publish pre-releases to crates.io and PyPI.
+/// Examples: `1.2.3`, `1.2.3-alpha.4`, `1.2.3-alpha.1+dev`.
 ///
-/// We use a `+githash` suffix for continuous pre-releases that you can download from our GitHub.
-/// We do NOT store that in this struct. See also `scripts/ci/version_util.py`.
+/// `-alpha.N+dev` versions are used for local or CI builds.
+/// `-alpha.N` versions are used for weekly releases.
+/// `-rc.N` versions are used for release candidates as we're preparing for a full release.
 ///
-/// The version numbers aren't allowed to be very large (current max: 31).
-/// This limited subset it chosen so that we can encode the version in 32 bits
+/// The version numbers (`N`) aren't allowed to be very large (current max: 31).
+/// This limited subset is chosen so that we can encode the version in 32 bits
 /// in our `.rrd` files and on the wire.
+///
+/// Here is the current binary format:
+/// ```text,ignore
+/// major    minor    patch    meta
+/// 00000000 00000000 00000000 00000000
+///                            ▲▲ ▲   ▲
+///                            ││ └┬──┘
+///                            ││  └─ N
+///                            │└─ rc/dev
+///                            └─ alpha
+/// ```
+///
+/// The valid bit patterns for `meta` are:
+/// - `100NNNNN` -> `-alpha.N`
+/// - `110NNNNN` -> `-alpha.N+dev`
+/// - `010NNNNN` -> `-rc.N`
+/// - `00000000` -> none of the above
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
 pub struct CrateVersion {
     major: u8,
     minor: u8,
     patch: u8,
-    alpha: Option<u8>,
-    prerelease: bool,
+    meta: Option<Meta>,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Meta {
+    Rc(u8),
+    Alpha(u8),
+    DevAlpha(u8),
+}
+
+impl Meta {
+    pub const fn to_byte(self) -> u8 {
+        match self {
+            Meta::Rc(value) => value | meta::RC,
+            Meta::Alpha(value) => value | meta::ALPHA,
+            Meta::DevAlpha(value) => value | meta::DEV_ALPHA,
+        }
+    }
+
+    pub const fn from_byte(v: u8) -> Option<Self> {
+        let tag = v & meta::TAG_MASK;
+        let value = v & meta::VALUE_MASK;
+        match tag {
+            meta::RC => Some(Self::Rc(value)),
+            meta::ALPHA => Some(Self::Alpha(value)),
+            meta::DEV_ALPHA => Some(Self::DevAlpha(value)),
+            _ => None,
+        }
+    }
+}
+
+const fn __slice(v: &[u8], start: Option<usize>, end: Option<usize>) -> &[u8] {
```

Comment:
```suggestion
const fn const_u8_slice_util(v: &[u8], start: Option<usize>, end: Option<usize>) -> &[u8] {
```

only because it's an impl detail doesn't mean it needs to have a name that let's the reviewer guess what it's for :P

Reply:
Sorry 😅